velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp

```diff
@@ -97,15 +102,38 @@ template <typename T, typename = void>
 struct AccumulatorTypeTraits {};
 
 template <typename T>
-struct AccumulatorTypeTraits<T, std::enable_if_t<isNumeric<T>(), void>> {
+struct AccumulatorTypeTraits<
+    T,
+    std::enable_if_t<isNumeric<T>() and !isDate<T>(), void>> {
   using AccumulatorType = T;
 };
 
 template <typename T>
-struct AccumulatorTypeTraits<T, std::enable_if_t<!isNumeric<T>(), void>> {
+struct AccumulatorTypeTraits<
+    T,
+    std::enable_if_t<!isNumeric<T>() and !isDate<T>(), void>> {
   using AccumulatorType = SingleValueAccumulator;
 };
 
+template <typename T>
```

Comment:
Move this code above, just immediately below the case for isNumeric at line 110 for better readability.